corehq/motech/migrations/0017_connectionsettings_use_aes_cbc_encryption.py

```diff
@@ -70,12 +71,59 @@ def reencrypted_last_token_with_cbc(connection):
         return ''
 
 
+def revert_api_settings(apps, schema_editor):
+    ConnectionSettings = apps.get_model("motech", "ConnectionSettings")
+
+    connect_settings_to_revert = ConnectionSettings.objects.exclude(
+        password__startswith=f'${ALGO_AES}$',
+        client_secret__startswith=f'${ALGO_AES}$',
+        last_token_aes__startswith=f'${ALGO_AES}$',
+    ).exclude(
+        password='',
+        client_secret='',
+        last_token_aes=''
+    )
+
+    for connection in connect_settings_to_revert:
+        connection.password = _revert_reencrypted_password_with_ecb(connection)
+        connection.client_secret = _revert_reencrypted_client_secret_with_ecb(connection)
+        connection.last_token_aes = _revert_reencrypted_last_token_with_ecb(connection)
+        connection.save()
+
+
+def _revert_reencrypted_password_with_ecb(connection):
+    if connection.password == '':
+        return ''
+    elif connection.password.startswith(f'${ALGO_AES_CBC}$'):
+        return reencrypt_cbc_to_ecb_mode(connection.password, f'${ALGO_AES_CBC}$')
+    else:
+        return connection.password
+
+
+def _revert_reencrypted_client_secret_with_ecb(connection):
+    if connection.client_secret == '':
+        return ''
+    elif connection.client_secret.startswith(f'${ALGO_AES_CBC}$'):
+        return reencrypt_cbc_to_ecb_mode(connection.client_secret, f'${ALGO_AES_CBC}$')
+    else:
+        return connection.client_secret
+
+
+def _revert_reencrypted_last_token_with_ecb(connection):
+    if connection.last_token_aes == '':
+        return ''
+    elif connection.last_token_aes.startswith(f'${ALGO_AES_CBC}$'):
+        return reencrypt_cbc_to_ecb_mode(connection.last_token_aes, f'${ALGO_AES_CBC}$')
+    else:
+        return connection.last_token_aes
```

Comment:
These three can be consolidated to one

Reply:
consolidated here [5c9b63a](https://github.com/dimagi/commcare-hq/pull/35641/commits/5c9b63afabaf9b37dfb76cc5077d824d71e89ca3)